rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,170 @@
+    if (
+        isinstance(block_identifier, int) and
+        block_identifier <= evm_inquirer.contract_multicall.deployed_block
+    ):
+        log.error(
+            f'No multicall contract at {evm_inquirer.chain_name} block {block_identifier}. '
+            f'{abi_name} query failed. Should implement direct queries',
+        )
+        return None
+
+    try:
+        output = evm_inquirer.multicall(
+            require_success=True,
+            calls=[(token.evm_address, contract.encode(method_name=method)) for method in methods],
+            block_identifier=block_identifier,
+        )
+    except (RemoteError, BlockchainQueryError) as e:
+        log.error(
+            f'Remote error calling {evm_inquirer.chain_name} multicall contract for {abi_name} '
+            f'token {token.evm_address} properties: {e!s}',
+        )
+        return None
+
+    # decode output
+    decoded = []
+    for (method_output, method_name) in zip(output, methods):
+        call_success = True
+        if call_success and len(method_output) != 0:
+            decoded_method = contract.decode(method_output, method_name)
+            # decoded_method is a tuple of decoded values. If it's only one then we just append that value. If it's more than one then we append the tuple. getReserves call returns a tuple of token reserves  # noqa: E501
+            decoded.append(decoded_method[0] if len(decoded_method) == 1 else decoded_method)
+        else:
+            log.debug(
+                f'{evm_inquirer.chain_name} multicall to {abi_name} failed to fetch field '
+                f'{method_name} for token {token.evm_address}',
+            )
+            return None
+
+    if len(decoded) < 4:
+        log.debug(
+            f'Unexpected number of decoded values ({len(decoded)}) while querying price from '
+            f'{evm_inquirer.chain_name} {abi_name} for token {token.evm_address}',
+        )
+        return None
+    elif len(decoded[3]) < 2:
+        log.debug(
+            f'Unexpected number of decoded pool reserves ({len(decoded[3])}) while querying '
+            f'price from {evm_inquirer.chain_name} {abi_name} for token {token.evm_address}',
+        )
+        return None
+
+    try:
+        token0 = EvmToken(evm_address_to_identifier(
+            address=decoded[0],
+            chain_id=token.chain_id,
+            token_type=EvmTokenKind.ERC20,
+        ))
+        token1 = EvmToken(evm_address_to_identifier(
+            address=decoded[1],
+            chain_id=token.chain_id,
+            token_type=EvmTokenKind.ERC20,
+        ))
+    except (UnknownAsset, WrongAssetType):
+        log.debug(
+            f'Unknown assets {decoded[0]} {decoded[1]} while querying price from '
+            f'{evm_inquirer.chain_name} {abi_name} for token {token.evm_address}',
+        )
+        return None
+
+    name = 'token supply'
+    location = 'uniswap-like pool price query'
```

Comment:
Remove them as function local variables since in esense they are constants.

Declare them in the top ofthe file with the `Final` modifier to mark them as constants and use them in the function

```suggestion
```